src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -128,6 +129,7 @@ public override void RemoveProject(ProjectId projectId)
             public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
             {
                 return e.Option.Feature == nameof(SimplificationOptions) ||
+                       e.Option.Feature == nameof(CodeStyleOptions) ||
```

Comment:
This seems fragile.  Option changing is rare right?  Why not be conservative and just reanalyze when options change?